File: 78-subsets/subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> ans=new ArrayList<>();
        List<Integer> ll=new ArrayList<>();
        subset(nums,ans,ll,0);
        return ans;
    }
    public void subset(int[] nums,List<List<Integer>> ans,List<Integer> ll,int idx){
        ans.add(new ArrayList<>(ll));
        for(int i=idx;i<nums.length;i++){
            ll.add(nums[i]);
            subset(nums,ans,ll,i+1);
            ll.remove(ll.get(ll.size()-1));
        }
    }
}

File: 78-subsets/subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans;
        int n=nums.size();
        int total=1<<n;
        for(int mask=0;mask<total;mask++){
            vector<int> ll;
            for(int i=0;i<n;i++){
                if(mask & (1<<i)){
                    ll.push_back(nums[i]);
                }
            }
            ans.push_back(ll);
        }
        return ans;
    }
   
};